src/triples/batch_random_ot.rs

```diff
@@ -72,6 +73,81 @@ pub async fn batch_random_ot_sender<C: CSCurve>(
     Ok((big_k0.try_into().unwrap(), big_k1.try_into().unwrap()))
 }
 
+pub async fn batch_random_ot_sender_many<C: CSCurve, const N: usize>(
+    ctx: Context<'_>,
+    mut chan: PrivateChannel,
+) -> Result<Vec<BatchRandomOTOutputSender>, ProtocolError> {
+    assert!(N > 0);
+    let mut big_yv = vec![];
+    let mut big_zv = vec![];
+    let mut yv = vec![];
+    for _ in 0..N {
+        // Spec 1
+        let y = C::Scalar::random(&mut OsRng);
+        let big_y = C::ProjectivePoint::generator() * y;
+        let big_z = big_y * y;
+        yv.push(y);
+        big_yv.push(big_y);
+        big_zv.push(big_z);
+    }
+    let mut big_y_affine_v = vec![];
+    for i in 0..N {
+        let big_y = &big_yv[i];
+        let big_y_affine = SerializablePoint::<C>::from_projective(&big_y);
+        big_y_affine_v.push(big_y_affine);
+    }
+    let wait0 = chan.next_waitpoint();
```

Comment:
I noticed that in the original code, this waitpoint comes before we calculate big_y_affine.  It doesn't look like this change will cause any issues, since there is nothing used from the waitpoint in teh calculation of big_y_affine.  

But i'm curious about the implications and why Lucas put it above the calculation, if it isn't needed.  Is it possible that this waitpoint is basically like a "confirm everyone else if participating" waitpoint?

Maybe this doesn't matter at all.  But it's odd that he would do this (wait, then calculate) if there's not a reason to do it this way.  Or maybe since this is the first waitpoint it doesn't matter at all - this waitpoint doesn't actually "wait" for anyone else, it's just grabbed since it's how we send the big_y_affine param.

Do you have any thoughts here?  O

Reply:
you are right, I overlooked this